packages/legend-application-studio/src/stores/sidebar-state/LocalChangesState.ts

```diff
@@ -428,66 +499,6 @@ export class LocalChangesState {
     }
   }
 
-  *refreshWorkspaceSyncStatus(): GeneratorFn<void> {
-    try {
-      this.refreshWorkspaceSyncStatusState.inProgress();
-      const currentRemoteRevision =
-        this.sdlcState.activeRemoteWorkspaceRevision;
-      yield flowResult(
-        this.sdlcState.fetchRemoteWorkspaceRevision(
-          this.sdlcState.activeProject.projectId,
-          this.sdlcState.activeWorkspace,
-        ),
-      );
-      if (
-        currentRemoteRevision.id !==
-        this.sdlcState.activeRemoteWorkspaceRevision.id
-      ) {
-        if (this.sdlcState.isWorkspaceOutOfSync) {
-          this.editorStore.localChangesState.workspaceSyncState.fetchIncomingRevisions();
-          const remoteWorkspaceEntities =
-            (yield this.editorStore.sdlcServerClient.getEntitiesByRevision(
-              this.sdlcState.activeProject.projectId,
-              this.sdlcState.activeWorkspace,
-              this.sdlcState.activeRemoteWorkspaceRevision.id,
-            )) as Entity[];
-          this.editorStore.changeDetectionState.workspaceRemoteLatestRevisionState.setEntities(
-            remoteWorkspaceEntities,
-          );
-          yield flowResult(
-            this.editorStore.changeDetectionState.workspaceRemoteLatestRevisionState.buildEntityHashesIndex(
-              remoteWorkspaceEntities,
-              LogEvent.create(
-                CHANGE_DETECTION_EVENT.CHANGE_DETECTION_LOCAL_HASHES_INDEX_BUILT,
-              ),
-            ),
-          );
-          yield flowResult(
-            this.editorStore.changeDetectionState.computeAggregatedWorkspaceRemoteChanges(),
-          );
-        } else {
-          this.editorStore.changeDetectionState.workspaceRemoteLatestRevisionState.setEntities(
-            [],
-          );
-          this.editorStore.changeDetectionState.setPotentialWorkspacePullConflicts(
-            [],
-          );
-          this.editorStore.changeDetectionState.setAggregatedWorkspaceRemoteChanges(
-            [],
-          );
-        }
-      }
-    } catch (error) {
-      assertErrorThrown(error);
-      this.editorStore.applicationStore.log.error(
-        LogEvent.create(LEGEND_STUDIO_APP_EVENT.SDLC_MANAGER_FAILURE),
-        error,
-      );
-    } finally {
-      this.refreshWorkspaceSyncStatusState.complete();
-    }
-  }
-
   *pushLocalChanges(pushMessage?: string): GeneratorFn<void> {
```

Comment:
I still think we can clean this state a bit more. 
I see a lot of repeated logic in both of the pushLocalChanges such as after committing and checking with sdlc. 
Maybe abstract speicifc actions inside pushLocalChanges and have each state implement those. 
like if for conflict like the conflict logic could be.

`processConflicts` where the form mode handles them and text mode doesn't. etc.